modules/status/sound_other.go

```diff
@@ -0,0 +1,60 @@
+// +build !openbsd,!freebsd,!netbsd
+
+package status
+
+import (
+	"log"
+
+	wmtheme "fyne.io/fynedesk/theme"
+	"github.com/mafik/pulseaudio"
+)
+
+// Destroy destroys the module
+func (b *sound) Destroy() {
+	if b.client == nil {
+		return
+	}
+	b.client.Close()
+}
+
+func (b *sound) value() (int, error) {
+	volume, err := b.client.Volume()
+	if err != nil {
+		return 0, err
+	}
+
+	return int(volume * 100), nil
+}
+
+func (b *sound) setup() error {
+	client, err := pulseaudio.NewClient()
+	if err != nil {
+		return err
+	}
+	b.client = client
+	return nil
+}
+
+func (b *sound) setValue(vol int) {
+	if err := b.client.SetVolume(float32(vol) / 100); err != nil {
+		log.Println("Failed to set volume", err)
+		return
+	}
+
+	b.bar.SetValue(float64(vol))
+}
+
+func (b *sound) toggleMute() {
+	toggl, err := b.client.ToggleMute()
+	if err != nil {
+		log.Println("toggleMute() failed", err)
+		return
+	}
+
+	if toggl {
+		b.mute.SetIcon(wmtheme.MuteIcon)
+	} else {
```

Comment:
perhaps could return early in the if and not have the else statement

Reply:
In this instance there are two different actions, 1 for muted and 1 for not.
Returning early is a good preference for handling edge cases when the body of the method is to come. In this instance both lines have equal importance.